compute_tools/src/compute.rs

```diff
@@ -1052,19 +1056,26 @@ impl ComputeNode {
         let pg_process = self.start_postgres(pspec.storage_auth_token.clone())?;
 
         let config_time = Utc::now();
-        if pspec.spec.mode == ComputeMode::Primary && !pspec.spec.skip_pg_catalog_updates {
-            let pgdata_path = Path::new(&self.pgdata);
-            // temporarily reset max_cluster_size in config
-            // to avoid the possibility of hitting the limit, while we are applying config:
-            // creating new extensions, roles, etc...
-            config::with_compute_ctl_tmp_override(pgdata_path, "neon.max_cluster_size=-1", || {
+        if pspec.spec.mode == ComputeMode::Primary {
+            if !pspec.spec.skip_pg_catalog_updates {
+                let pgdata_path = Path::new(&self.pgdata);
+                // temporarily reset max_cluster_size in config
+                // to avoid the possibility of hitting the limit, while we are applying config:
+                // creating new extensions, roles, etc...
+                config::with_compute_ctl_tmp_override(
+                    pgdata_path,
+                    "neon.max_cluster_size=-1",
+                    || {
+                        self.pg_reload_conf()?;
+
+                        self.apply_config(&compute_state)?;
+
+                        Ok(())
+                    },
+                )?;
                 self.pg_reload_conf()?;
-
-                self.apply_config(&compute_state)?;
-
-                Ok(())
-            })?;
-            self.pg_reload_conf()?;
+            }
+            self.post_apply_config()?;
```

Comment:
This changes the behaviour so that `post_apply_config()` is executed even if `skip_catalog_updates==true`. That means we will always perform `alter extension neon update`, too. Is that intentional.

In general, I don't like adding automatic operations like this to startup. Do we really need to log all the extensions at every startup? ISTM that having the the API endpoint to query them is sufficient.

Reply:
The idea was to make this logging v0 version to start getting the statistics, until we have implemented the API caller part.

As for `post_apply_config`, it was like this [pre-v17](https://github.com/neondatabase/neon/blob/842be0ba74c4c6e4245c29c3fffae4401d282c4a/compute_tools/src/compute.rs#L1074) (this is the commit before v17), we just lost this code during v17 merge. I am just bringing it back, see the commit message.

Reply:
>The idea was to make this logging v0 version to start getting the statistics, until we have implemented the API caller part.

Hmm, then we'll need to implement log scraping to get this. I'd prefer to just wait for the API caller to be honest.  It sounds like it will be only a small amount of work.

If we add logging now, we'll need to remember to rip out out later.

Reply:
> As for post_apply_config, it was like this [pre-v17](https://github.com/neondatabase/neon/blob/842be0ba74c4c6e4245c29c3fffae4401d282c4a/compute_tools/src/compute.rs#L1074) (this is the commit before v17), we just lost this code during v17 merge. I am just bringing it back, see the commit message.

Gotcha. Maybe extract that to a separate PR?

Reply:
https://github.com/neondatabase/neon/pull/9220